### ISSUE TITLE
 [NODE-OPCUA-E11] message size 16385 exceeds the negotiated message size 16384 nb chunks 1.

### ISSUE THREAD

**Issue by szelski** [node-opcua/node-opcua#1335]
Problem:
I cannot establish connection to a certain OPC UA Server with my OPC UA Client. (Connections to other tested OPC UA Servers do work)
I have a similar Issue like #1263 which is already closed. But the replys there, which i tested could not help me. In my case there seems to be a mismatch of 1 Byte only: 

[NODE-OPCUA-E11] message size 16385 exceeds the negotiated message size 16384 nb chunks 1

Details (process.env.NODEOPCUADEBUG = "TRANSPORT{HELACK-CHUNK}";):

10:17:38.521Z :client_tcp_transport          :285   sending Hello
 ...                                                 protocolVersion   = 0
 ...                                                receiveBufferSize = 655360
 ...                                                sendBufferSize    = 655360
 ...                                                maxMessageSize    = 0
 ...                                                maxChunkCount     = 0
 ...                                                endpointUrl       = opc.tcp://172.18.4.25:48020
 ...
10:17:38.523Z :client_tcp_transport          :263   receiving Ack
 ...                                                 protocolVersion   = 0
 ...                                                receiveBufferSize = 16384
 ...                                                sendBufferSize    = 16384
 ...                                                maxMessageSize    = 16384
 ...                                                maxChunkCount     = 1
 ...
10:17:38:524    MESSAGE BUILDER LIMITS maxMessageSize =  16384 maxChunkCount =  1 maxChunkSize =  16384 ( 16384 )
10:17:38.527Z :message_chunker               :117   [NODE-OPCUA-E11] message size 16385 exceeds the negotiated message size 16384 nb chunks 1

**Comment by erossignon** [comment]
Please forward this request to nodered-contrib-opcua forum, if you need a community answer.


Alternatively, we encourage you to use [`@opcua/for-node-red`](https://flows.nodered.org/node/@opcua/for-node-red) ,  a industrial-grade OPC UA node for node-red developed and supported by the NodeOPCUA team at Sterfive.

**Comment by szelski** [comment]
Why you just rename my Issue like that?
I do not have anything to do with node-red!

**Comment by erossignon** [comment]
It was mentioned the issue was similar to #1263, which relates to node-red-contrib-opcua. Hence the confusion. Sorry for this. 

Please adjust the initial issue description by filling carefully the Bug reporting template , providing  clear step by step instructions to reproduce, version of node-opcua involved, describing the third party system etc ....

**Comment by szelski** [comment]
-->

**Current behavior**

I have an OPC UA Client created with the node-opcua library. The connection to other OPC UA servers generally works.
But with a specific device (labeling system), this connection does not work. However,
there are other OPC UA clients that can connect to this labeling system, as I have tested myself.
The connection works with the OPC UA client "UaExpert" from Unified Automation, and you can browse through the OPC UA structure of the labeling system.

This is evidence that an OPC UA client-server connection with the labeling system generally works.

**Describe the bug**

First, I have enabled the following for an extended debug output.

```javascript
process.env.NODEOPCUADEBUG = "TRANSPORT{HELACK-CHUNK}";
```

Then I used this connection routine.

```javascript
this.client = opcua.OPCUAClient.create({
  endpointMustExist: true, // also tried false and different transportSettings{...}
});
```

I received the following outputs.

´´´
10:17:38.521Z :client_tcp_transport :285 sending Hello
... protocolVersion = 0
... receiveBufferSize = 655360
... sendBufferSize = 655360
... maxMessageSize = 0
... maxChunkCount = 0
... endpointUrl = opc.tcp://172.18.4.25:48020
...
10:17:38.523Z :client_tcp_transport :263 receiving Ack
... protocolVersion = 0
... receiveBufferSize = 16384
... sendBufferSize = 16384
... maxMessageSize = 16384
... maxChunkCount = 1
...
10:17:38:524 MESSAGE BUILDER LIMITS maxMessageSize = 16384 maxChunkCount = 1 maxChunkSize = 16384 ( 16384 )
10:17:38.527Z :message_chunker :117 [NODE-OPCUA-E11] message size 16385 exceeds the negotiated message size 16384 nb chunks 1
´´´

Subsequently, no connection is established.

As I understand it, the client and server agree on the server's maxMessageSize of 16384 bytes. However, the client violates this agreement by sending a message with 16385 bytes (one byte more than agreed upon).

**Step by step instruction to reproduce**

Actually i followed the advice:

https://github.com/node-opcua/node-opcua/issues/1263#issuecomment-1502928573

Steps to reproduce the behavior:

1.  Activate:

```javascript
process.env.NODEOPCUADEBUG = "TRANSPORT{HELACK-CHUNK}";
```

2.  Try to connect:

```javascript
this.client = opcua.OPCUAClient.create({
  endpointMustExist: true, // also tried false and different transportSettings{...}
});
```

3. Analyze Debug Output

**Expected behavior**

In the positive case, I would expect a successful connection setup from the client to the server.

**Screenshots**



**Gist**

```javascript
process.env.NODEOPCUADEBUG = "TRANSPORT{HELACK-CHUNK}";
const { OPCUAClient } = require("node-opcua");

// replace this line with the targeted server endpoint URL:
const endpointUrl = "opc.tcp://...";
(async () => {
  const client = OPCUAClient.create({});
  await client.connect(endpointUrl);
  await client.disconnect();
})();
```

**Context**

- (x) my request is related to node-opcua acting as an OPCUA CLIENT
- ( ) my request is related to node-opcua acting as an OPCUA SERVER

- ( ) I have installed node-opcua from source ( using git clone)
- (x) I have installed node-opcua as a package ( using npm install )
- ( ) I am using an application that uses node-opcua

  - ( ) node-red
  - ( ) other : please specify

- Device: **\_\_**
- OS version: **\_\_**

  - (x) Windows : version : **\_Microsoft Windows 10 Pro: 10.0.19045 Nicht zutreffend Build 19045\_**
  - ( ) Linux : version : **\_\_**
  - ( ) MacOs : version : **\_\_**
  - ( ) Raspbian: version : **\_\_**
  - ( ) Other : specify : **\_\_**

- Description of the other OPCUA system I am trying to connect to:

  - Name: **\_\_**
  - Version: **\_\_**
  - Manufacturer/Software vendor: **\_\_**
  - link to the manufacturer or product site: https://

- **node-opcua version:** :
  {
  "name": "node-opcua",
  "version": "2.113.0",
  "description": "pure nodejs OPCUA SDK - module node-opcua",
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "scripts": {
  "clean": "npx rimraf node_modules dist \*.tsbuildinfo",
  "test": "mocha",
  "build": "tsc -b"
  },
  "engines": {
  "node": ">=8.10"
  },
  "dependencies": {
  "@types/semver": "^7.5.1",
  "chalk": "4.1.2",
  "node-opcua-address-space": "2.113.0",
  "node-opcua-address-space-for-conformance-testing": "2.113.0",
  "node-opcua-aggregates": "2.113.0",
  "node-opcua-assert": "2.105.0",
  "node-opcua-basic-types": "2.113.0",
  "node-opcua-binary-stream": "2.110.0",
  "node-opcua-certificate-manager": "2.113.0",
  "node-opcua-client": "2.113.0",
  "node-opcua-client-crawler": "2.113.0",
  "node-opcua-client-proxy": "2.113.0",
  "node-opcua-common": "2.113.0",
  "node-opcua-constants": "2.98.1",
  "node-opcua-crypto": "4.5.0",
  "node-opcua-data-access": "2.113.0",
  "node-opcua-data-model": "2.113.0",
  "node-opcua-data-value": "2.113.0",
  "node-opcua-debug": "2.113.0",
  "node-opcua-enum": "2.110.0",
  "node-opcua-factory": "2.113.0",
  "node-opcua-hostname": "2.105.0",
  "node-opcua-nodeid": "2.113.0",
  "node-opcua-nodesets": "2.110.0",
  "node-opcua-numeric-range": "2.113.0",
  "node-opcua-packet-analyzer": "2.113.0",
  "node-opcua-secure-channel": "2.113.0",
  "node-opcua-server": "2.113.0",
  "node-opcua-server-discovery": "2.113.0",
  "node-opcua-service-browse": "2.113.0",
  "node-opcua-service-call": "2.113.0",
  "node-opcua-service-discovery": "2.113.0",
  "node-opcua-service-endpoints": "2.113.0",
  "node-opcua-service-filter": "2.113.0",
  "node-opcua-service-history": "2.113.0",
  "node-opcua-service-node-management": "2.113.0",
  "node-opcua-service-query": "2.113.0",
  "node-opcua-service-read": "2.113.0",
  "node-opcua-service-register-node": "2.113.0",
  "node-opcua-service-secure-channel": "2.113.0",
  "node-opcua-service-session": "2.113.0",
  "node-opcua-service-subscription": "2.113.0",
  "node-opcua-service-translate-browse-path": "2.113.0",
  "node-opcua-service-write": "2.113.0",
  "node-opcua-status-code": "2.110.0",
  "node-opcua-transport": "2.113.0",
  "node-opcua-types": "2.113.0",
  "node-opcua-utils": "2.110.0",
  "node-opcua-variant": "2.113.0",
  "node-opcua-vendor-diagnostic": "2.113.0",
  "semver": "^7.5.4"
  },
  "devDependencies": {
  "should": "^13.2.3"
  },
  "author": "Etienne Rossignon",
  "license": "MIT",
  "repository": {
  "type": "git",
  "url": "git://github.com/node-opcua/node-opcua.git"
  },
  "keywords": [
  "OPCUA",
  "opcua",
  "m2m",
  "iot",
  "opc ua",
  "internet of things"
  ],
  "funding": {
  "url": "https://github.com/sponsors/erossignon"
  },
  "homepage": "http://node-opcua.github.io/",
  "gitHead": "36db335391fedd39726990a1b37f7768da16466a",
  "files": [
  "dist",
  "source"
  ]
  }

- **Node:**

  `node --version` = v20.5.1

**Comment by szelski** [comment]
The other OPCUA system I am trying to connect to:

Name: Alpha HSM (https://www.bluhmsysteme.com/etikettierung/etikettenspender/alpha-hsm.html)
Version: (have to look up later)
Manufacturer/Software vendor: https://s-q-f.com/

So you want to buy this system to reproduce the problem?

I think the Debug log of the connection process that i provided is the most important information.

**Comment by erossignon** [comment]
I should be able to investigate without .... stay tune

**Comment by erossignon** [comment]
fixed in [node-opcua@2.200.0](https://github.com/node-opcua/node-opcua/releases/tag/v2.120.0)

**Comment by szelski** [comment]
Thanks a lot, i tested it and it works now!
